lib/stream.js

```diff
@@ -24,6 +24,7 @@
 const pipeline = require('internal/streams/pipeline');
 const eos = require('internal/streams/end-of-stream');
 const internalBuffer = require('internal/buffer');
+const promises = require('internal/streams/promises');
```

Comment:
I would avoid requiring this here and load it only if needed. This adds another full level of circular dependencies and loading that we should really avoid.

Reply:
We can lazy load promises. https://github.com/nodejs/node/pull/33991#discussion_r443200195